Mac OS and Windows support

Mac OS support is still missing because there is only a simple `Makefile` in `lrslib` instead of a `configure` file that generates it. Windows support is missing because there is no lrslib package in [this repo](https://build.opensuse.org/project/show/windows:mingw:win32) and there need to be one containing the changes I made to lrslib in my fork.

Jan Engelhardt (@jengelh) has made a patch for `lrslib` to make it use GNU automake [here](https://build.opensuse.org/package/show/science/lrslib).
I have submitted the changes of the lrslib fork up to [this commit](https://github.com/blegat/lrslib/commit/89bf29524143836a298e3f7f513e8b96d6356031) (which is currently master) to David Avis to make it mainstream.

The plan to finish Linux support is as follows:
- [ ] once David Avis has released the first round of changes, use the upstream version for Linux build.

The plan for adding Mac OS support is as follows:
- [ ] apply Jan Engelhardt's patch to the [lrslib fork](https://github.com/blegat/lrslib)
- [ ] use the fork version for Mac OS build
- [ ] once David Avis has released the first round of changes, suggest to add Jan Engelhardt's patch
- [ ] once he has released them, use the mainstream version for Mac OS

The plan for adding Windows support is as follows:
- [ ] once David Avis has release the first round of changes, bump the version number used [here](https://build.opensuse.org/package/show/home:jengelh:dev/lrslib)
- [ ] from this build, make a Windows cross compilation build and submit it to [this repo](https://build.opensuse.org/project/show/windows:mingw:win32)
- [ ] Use this with [WinRPM](https://github.com/JuliaPackaging/WinRPM.jl)

@oyamad Thanks for the notebook. I don't see what is causing an `InexactError` at line 40 of `conversion.jl`, could you dig down the conversion that is causing the failure ? Is it in `convertoutput` ?
For the last issue `AssertionError: ispoint == zero(Rational{BigInt}) || ispoint == one(Rational{BigInt})`, could you give me what is printed if you add
```
if !(ispoint == zero(Rational{BigInt}) || ispoint == one(Rational{BigInt}))
    @show ispoint
end
```
just before the `@assert` ?

`ispoint = -2//1` is the output of `@show ispoint`:

```jl
julia> using Polyhedra

julia> using LRSLib

julia> lib = LRSLibrary();

julia> A = [1 1 1; 1 0 0; 0 1 0; 0 0 1; -1 0 0; 0 -1 0; 0 0 -1];

julia> b = [6, 3, 3, 3, -1, -1, -1];

julia> linset = IntSet([1]);

julia> V = [2 3 1; 1 3 2; 3 1 2; 3 2 1; 2 1 3; 1 2 3];

julia> ine = SimpleHRepresentation(A, b, linset);

julia> poly = polyhedron(ine, lib);

julia> detecthlinearities!(poly)
ERROR: InexactError()
 in schedule_and_wait(::Task, ::Void) at ./event.jl:110
 in consume(::Task) at ./task.jl:269
 in done at ./task.jl:274 [inlined]
 in enumtomat(::LRSLib.LRSGeneratorMatrix{3}) at /Users/oyama/.julia/v0.5/LRSLib/src/conversion.jl:40
 in convert(::Type{Polyhedra.LiftedHRepresentation{3,Rational{BigInt}}}, ::LRSLib.LRSGeneratorMatrix{3}) at /Users/oyama/.julia/v0.5/LRSLib/src/conversion.jl:48
 in getine(::LRSLib.LRSPolyhedron{3}) at /Users/oyama/.julia/v0.5/LRSLib/src/polyhedron.jl:43
 in detecthlinearities!(::LRSLib.LRSPolyhedron{3}) at /Users/oyama/.julia/v0.5/LRSLib/src/polyhedron.jl:154

julia> LRSLib.getextm(poly, :Fresh)
V-representation
begin
 6 4 rational
ispoint = -2//1
Error showing value of type LRSLib.LRSGeneratorMatrix{3}:
ERROR: AssertionError: ispoint == zero(Rational{BigInt}) || ispoint == one(Rational{BigInt})
 in extractrow(::LRSLib.LRSGeneratorMatrix{3}, ::Int64) at /Users/oyama/.julia/v0.5/LRSLib/src/matrix.jl:269
 in nextvrep(::LRSLib.LRSGeneratorMatrix{3}, ::Int64) at /Users/oyama/.julia/v0.5/LRSLib/src/matrix.jl:208
 in next(::Polyhedra.VRepIterator{3,Rational{BigInt},3,Rational{BigInt}}, ::Tuple{Int64,Int64}) at /Users/oyama/.julia/v0.5/Polyhedra/src/representation.jl:109
 in show(::IOContext{Base.Terminals.TTYTerminal}, ::LRSLib.LRSGeneratorMatrix{3}) at /Users/oyama/.julia/v0.5/Polyhedra/src/show.jl:38
 in display(::Base.REPL.REPLDisplay{Base.REPL.LineEditREPL}, ::MIME{Symbol("text/plain")}, ::LRSLib.LRSGeneratorMatrix{3}) at ./REPL.jl:132
 in display(::Base.REPL.REPLDisplay{Base.REPL.LineEditREPL}, ::LRSLib.LRSGeneratorMatrix{3}) at ./REPL.jl:135
 in display(::LRSLib.LRSGeneratorMatrix{3}) at ./multimedia.jl:143
 in print_response(::Base.Terminals.TTYTerminal, ::Any, ::Void, ::Bool, ::Bool, ::Void) at ./REPL.jl:154
 in print_response(::Base.REPL.LineEditREPL, ::Any, ::Void, ::Bool, ::Bool) at ./REPL.jl:139
 in (::Base.REPL.##22#23{Bool,Base.REPL.##33#42{Base.REPL.LineEditREPL,Base.REPL.REPLHistoryProvider},Base.REPL.LineEditREPL,Base.LineEdit.Prompt})(::Base.LineEdit.MIState, ::Base.AbstractIOBuffer{Array{UInt8,1}}, ::Bool) at ./REPL.jl:652
 in run_interface(::Base.Terminals.TTYTerminal, ::Base.LineEdit.ModalInterface) at ./LineEdit.jl:1579
 in run_interface(::Base.Terminals.TTYTerminal, ::Base.LineEdit.ModalInterface) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in run_frontend(::Base.REPL.LineEditREPL, ::Base.REPL.REPLBackendRef) at ./REPL.jl:903
 in run_repl(::Base.REPL.LineEditREPL, ::Base.##930#931) at ./REPL.jl:188
 in _start() at ./client.jl:360
 in _start() at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
```

Regarding conversion, what shall I check?

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -18,6 +18,9 @@ jobs:
           - version: '1'
             os: macOS-latest
             arch: x64
+          - version: '1'
+            os: windows-latest
+            arch: x64
 # See https://github.com/JuliaPolyhedra/LRSLib.jl/runs/2705442631?check_suite_focus=true#step:6:135
 #          - version: '1'
 #            os: ubuntu-latest
@@ -27,6 +30,9 @@ jobs:
             arch: x64
           - version: 'nightly'
             os: macOS-latest
+            arch: x64
+          - version: 'nightly'
+            os: windows-latest
             arch: x64
 #          - version: 'nightly'
 #            os: ubuntu-latest

--- a/README.md
+++ b/README.md
@@ -13,8 +13,6 @@ As written in the [user guide of lrs](http://cgm.cs.mcgill.ca/~avis/C/lrslib/USE
 I have [forked lrs](https://github.com/blegat/lrslib) to add a few functions to help doing the wrapper.
 These changes are not upstream yet so this version is used instead of the upstream version.
 
-**Important notice**: Windows is not supported yet.
-
 [build-img]: https://github.com/JuliaPolyhedra/LRSLib.jl/workflows/CI/badge.svg?branch=master
 [build-url]: https://github.com/JuliaPolyhedra/LRSLib.jl/actions?query=workflow%3ACI
 [codecov-img]: http://codecov.io/github/JuliaPolyhedra/LRSLib.jl/coverage.svg?branch=master

--- a/src/matrix.jl
+++ b/src/matrix.jl
@@ -104,7 +104,7 @@ mutable struct HMatrix <: Polyhedra.MixedHRep{Rational{BigInt}}
     Q::Ptr{Clrs_dat}
     status::Symbol
     lin::Union{Nothing, LRSLinearitySpace}
-    function HMatrix(N::Int, P::Ptr{Clrs_dic}, Q::Ptr{Clrs_dat})
+    function HMatrix(N::Integer, P::Ptr{Clrs_dic}, Q::Ptr{Clrs_dat})
         @assert N == getd(P)
         m = new(N, P, Q, :AtNoBasis, nothing)
         finalizer(myfree, m)
@@ -120,7 +120,7 @@ mutable struct VMatrix <: Polyhedra.MixedVRep{Rational{BigInt}}
     status::Symbol
     lin::Union{Nothing, LRSLinearitySpace}
     cone::Bool # If true, LRS will not return any point so we need to add the origin
-    function VMatrix(N::Int, P::Ptr{Clrs_dic}, Q::Ptr{Clrs_dat})
+    function VMatrix(N::Integer, P::Ptr{Clrs_dic}, Q::Ptr{Clrs_dat})
         @assert N + 1 == getd(P)
         m = _length(P)
         cone = !iszero(m) # If there is no ray and no point, it is empty so we should not add the origin

--- a/src/nash.jl
+++ b/src/nash.jl
@@ -279,7 +279,7 @@ function nashsolve(filename1::AbstractString, filename2::AbstractString)
 
     # From lrs_solve_nash_legacy
     if Q2.nlinearity > 0
-        ccall((:free, liblrs), Cvoid, (Ptr{Cvoid},), Q2.linearity)
+        ccall(:free, Cvoid, (Ptr{Cvoid},), Q2.linearity)
     end
     ptr = @lrs_ccall2(
         xcalloc, Ptr{Cvoid}, (Clong, Clong, Clong, Cstring),

